core/src/serde.rs

```diff
@@ -260,16 +260,46 @@ impl TryFrom<BootstrapConfigSerde> for BootstrapConfig {
     type Error = SerdeConversionError;
 
     fn try_from(c: BootstrapConfigSerde) -> Result<Self, Self::Error> {
-        let pool_contract_parameters =
-            PoolContractParameters::try_from(c.pool_contract_parameters)?;
+        let pool_contract_parameters = PoolContractParameters::build_with(
+            base16::decode(c.pool_contract_parameters.ergo_tree_bytes.as_str())?,
+            c.pool_contract_parameters.refresh_nft_index,
+            c.pool_contract_parameters.update_nft_index,
+        )?;
         let refresh_contract_parameters =
-            RefreshContractParameters::try_from(c.refresh_contract_parameters)?;
-        let update_contract_parameters =
-            UpdateContractParameters::try_from(c.update_contract_parameters)?;
-        let ballot_contract_parameters =
-            BallotContractParameters::try_from(c.ballot_contract_parameters)?;
-        let oracle_contract_parameters =
-            OracleContractParameters::try_from(c.oracle_contract_parameters)?;
```

Comment:
After removing these `try_from` calls, the only other place that calls them are here: https://github.com/ergoplatform/oracle-core/blob/12c065a3f38124fd855e30a84a5d641609d56f7f/core/src/serde.rs#L132-L155

Perhaps we can remove the `TryFrom` impls altogether? It's clearer for the reader to see that `build_with/checked_load` was called instead.

Reply:
Good call! With the introduction of `checked_load` and `build_with` these `TryFrom` implementations do not convey the actual operation they are doing. I thought to leave `TryFrom` as `checked_load` cases only, but calling `checked_load` is a better option.